minecraft/realms/realms.go

```diff
@@ -0,0 +1,213 @@
+package realms
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sandertv/gophertunnel/minecraft/auth"
+	"golang.org/x/oauth2"
+)
+
+// RealmsApi is an instance of the realms api with a token
+type RealmsApi struct {
+	token_src  oauth2.TokenSource
+	xbox_token *auth.XBLToken
```

Comment:
For consistency, you might want to change this to camel case and rename these fields to `tokenSrc` and `xboxToken`.